### DISCUSSION TITLE
imagemagick output strange on png when piping from Terminal

### DISCUSSION THREAD

**Discussion by neural-oD** [ImageMagick/ImageMagick#3841] [Help]
I'm stuck with something. I had a previous question found here at stack exchange:
[Previous question][1]

So bascally the solution that was there, for no apparent reason stopped working. Only thing that might have happened was apt update commands. Nothing else has changed.
So basically further up the script I have this:
```
faketty() {
          script -qfc "$(printf "%q " "$@")" /dev/null
           }
```
Now further down I have this: 

```
faketty sudo -u $usr wp --path="${htmldir}"/ plugin list --all --format=table|awk '/name/,EOF'|tee >(convert -font Courier -pointsize 18 label:@- "${savdir}${url}_${curr_date}.list.png")```
```
which renders correctly on the terminal, and the png file:
[![enter image description here][2]][2]

However the next section needs to update the plugins:
```
faketty sudo -u $usr wp --path="${htmldir}"/ plugin update --all |awk '/Disabling\ Maintenance/,EOF'|tee >(convert -font Courier -pointsize 18 label:@- "${savdir}${url}_${curr_date}.plugins.png")
```

This one renders correctly on the terminal:
[![enter image description here][3]][3]

but not the output png file:
[![enter image description here][4]][4]

  [1]: https://wordpress.stackexchange.com/questions/386665/formatting-messed-up-when-piping-wp-commands
  [2]: https://i.stack.imgur.com/R3T0I.png
  [3]: https://i.stack.imgur.com/wWvJD.png
  [4]: https://i.stack.imgur.com/LK1gI.png

I'm not sure why this is doing this, and working on the listing of the plugins, but not working on the updating plugins.

Any help and suggestions would be appreciated.

here is my convert version info:
`Version: ImageMagick 6.9.7-4 Q16 x86_64 20170114 http://www.imagemagick.org                                                                                                                                        
Copyright: © 1999-2017 ImageMagick Studio LLC                                                                                                                                                                      
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP 
Delegates (built-in): bzlib djvu fftw fontconfig freetype jbig jng jpeg lcms lqr ltdl lzma openexr pangocairo png tiff wmf x xml zlib
`

Thanks

**Comment by snibgo** [comment]
As far as I can see, your question is about the text that is piped to convert's "label:", rather than about what "convert" is doing. Is that right?

I know nothing about the tools that generate that text, and can't help, sorry.

**Reply by neural-oD** [reply]
Hi @snibgo 
Thanks for the reply. Basically the code is only splitting the pipe, so that it's sent to "convert". As you can see the one renders fine, the other spits out what looks like a hex-editor view of the contents?
I'm trying to resolve where this is happening, maybe I should reach out to the wp-cli team, and see what they say?
